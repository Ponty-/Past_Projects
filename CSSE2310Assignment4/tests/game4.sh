#!/bin/bash

if [ $# != 4 ]
then
    echo "Usage: $0 binarydir port resultdir correct_dir" 
    exit 1
fi



bindir=$1
port=$2
res=$3
right=$4
name=$0

expected=0

if [ ! -f $bindir/client499 ]
then
    echo "$bindir/client499 is missing"
    exit 1
fi

if [ ! -f $bindir/serv499 ]
then
    echo "$bindir/serv499 is missing"
    exit 1
fi



$bindir/serv499 $port '%s%s%s%s%s' 4.deck &
spid=$!
$bindir/client499 zzzzzzzz gg $port > $res/$name.a.out 2>$res/$name.a.err < ax &
pid1=$!
$bindir/client499 123456 gg $port > $res/$name.b.out 2>$res/$name.b.err < bx &
pid2=$!
$bindir/client499 CCC gg $port > $res/$name.c.out 2>$res/$name.c.err < cx &
pid3=$!
$bindir/client499 dddd gg $port > $res/$name.d.out 2>$res/$name.d.err < dx &
pid4=$!

wait $pid1 $pid2 $pid3 $pid4 2> /dev/null

$bindir/client499 zzzzA game $port > $res/$name.a.2.out 2>$res/$name.a.2.err < ax &
pid1=$!
$bindir/client499 123B game $port > $res/$name.b.2.out 2>$res/$name.b.2.err < bx &
pid2=$!
$bindir/client499 C game $port > $res/$name.c.2.out 2>$res/$name.c.2.err < cx &
pid3=$!
$bindir/client499 D game $port > $res/$name.d.2.out 2>$res/$name.d.2.err < dx &
pid4=$!

wait $pid1 $pid2 $pid3 $pid4 2> /dev/null



kill $spid 2> /dev/null
wait $spid 2> /dev/null

#if [ $result != $expected ]
#then
#    echo "Expected $expected got $result"
#    exit 1
#fi

for v in a b c d
do

diff -q $res/$name.$v.2.out $right/$name.$v.2.out > /dev/null
if [ $? != 0 ]
then
   echo "Stdout mismatch for client $v"
   echo "diff $res/$name.$v.2.out $right/$name.$v.2.out"
   exit 2
fi

diff -q $res/$name.$v.2.err $right/$name.$v.2.err > /dev/null
if [ $? != 0 ]
then
   echo "Stderr mismatch for client $v"
   echo "diff $res/$name.$v.2.err $right/$name.$v.2.err"
   exit 3
fi

done

echo OK
exit 0

