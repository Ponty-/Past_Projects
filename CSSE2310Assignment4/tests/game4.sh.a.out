Info: %s%s%s%s%s
Info: Team1: 123456, dddd
Info: Team2: CCC, zzzzzzzz
S: 9 4
C: K 9 4
D: A Q 8 4 2
H: J 8 5
Info: 123456 bids 7H
Info: CCC passes
Info: dddd bids 8S
[8S] - Bid (or pass)> Info: 123456 passes
Info: dddd plays TS
S: 9 4
C: K 9 4
D: A Q 8 4 2
H: J 8 5
[S] play> Info: 123456 plays AS
Info: CCC plays 7S
Info: 123456 won
Info: 123456 plays KS
Info: CCC plays QS
Info: dddd plays 6S
S: 4
C: K 9 4
D: A Q 8 4 2
H: J 8 5
[S] play> Info: 123456 won
Info: 123456 plays QH
Info: CCC plays AH
Info: dddd plays 4H
S:
C: K 9 4
D: A Q 8 4 2
H: J 8 5
[H] play> Info: CCC won
Info: CCC plays QC
Info: dddd plays 2C
S:
C: K 9 4
D: A Q 8 4 2
H: J 8
[C] play> Info: 123456 plays 5C
Info: CCC won
Info: CCC plays KD
Info: dddd plays 3D
S:
C: K 9
D: A Q 8 4 2
H: J 8
[D] play> S:
C: K 9
D: A Q 8 4 2
H: J 8
[D] play> Info: 123456 plays JS
Info: 123456 won
Info: 123456 plays 7H
Info: CCC plays TH
Info: dddd plays 9H
S:
C: K 9
D: A Q 8 4
H: J 8
[H] play> Info: CCC won
Info: CCC plays 7C
Info: dddd plays AC
S:
C: K 9
D: A Q 8 4
H: J
[C] play> Info: 123456 plays 8C
Info: dddd won
Info: dddd plays KH
S:
C: K
D: A Q 8 4
H: J
[H] play> S:
C: K
D: A Q 8 4
H: J
[H] play> Info: 123456 plays 3H
Info: CCC plays 2H
Info: dddd won
Info: dddd plays TC
S:
C: K
D: A Q 8 4
H:
[C] play> Info: 123456 plays JC
Info: CCC plays 3C
Info: zzzzzzzz won
S:
C:
D: A Q 8 4
H:
Lead> Info: 123456 plays 2S
Info: CCC plays 5D
Info: dddd plays JD
Info: 123456 won
Info: 123456 plays 8S
Info: CCC plays 7D
Info: dddd plays 6C
S:
C:
D: Q 8 4
H:
[S] play> Info: 123456 won
Info: 123456 plays 3S
Info: CCC plays TD
Info: dddd plays 6D
S:
C:
D: 8 4
H:
[S] play> Info: 123456 won
Info: 123456 plays 5S
Info: CCC plays 6H
Info: dddd plays 9D
S:
C:
D: 8
H:
[S] play> Info: 123456 won
Info: Team 1=220, Team 2=0
S: 9 4
C: K 7 4
D: A Q 8 4 2
H: J 8 5
Info: 123456 bids 6S
Info: CCC bids 6H
Info: dddd passes
[6H] - Bid (or pass)> Info: 123456 passes
Info: CCC plays AH
Info: dddd plays KH
S: 9 4
C: K 7 4
D: A Q 8 4 2
H: J 8 5
[H] play> Info: 123456 plays 3H
Info: CCC won
Info: CCC plays KD
Info: dddd plays 3D
S: 9 4
C: K 7 4
D: A Q 8 4 2
H: J 8
[D] play> Info: 123456 plays 2S
Info: zzzzzzzz won
S: 9 4
C: K 7 4
D: Q 8 4 2
H: J 8
Lead> Info: 123456 plays 5C
Info: CCC plays 3C
Info: dddd plays 2C
Info: zzzzzzzz won
S: 9 4
C: 7 4
D: Q 8 4 2
H: J 8
Lead> Info: 123456 plays TC
Info: CCC plays 5D
Info: dddd plays 6D
Info: zzzzzzzz won
S: 9 4
C: 7 4
D: 8 4 2
H: J 8
Lead> Info: 123456 plays AS
Info: CCC plays 7S
Info: dddd plays 6S
Info: 123456 won
Info: 123456 plays KS
Info: CCC plays QS
Info: dddd plays TS
S: 4
C: 7 4
D: 8 4 2
H: J 8
[S] play> Info: 123456 won
Info: 123456 plays 3S
Info: CCC plays TH
Info: dddd plays 6C
S:
C: 7 4
D: 8 4 2
H: J 8
[S] play> Info: CCC won
Info: CCC plays QC
Info: dddd plays AC
S:
C: 7 4
D: 8 4
H: J 8
[C] play> Info: 123456 plays JC
Info: dddd won
Info: dddd plays 4H
S:
C: 7
D: 8 4
H: J 8
[H] play> Info: 123456 plays 7H
Info: CCC plays 2H
Info: zzzzzzzz won
S:
C: 7
D: 8 4
H: J
Lead> Info: 123456 plays QH
Info: CCC plays 6H
Info: dddd plays 9H
Info: 123456 won
Info: 123456 plays 5S
Info: CCC plays 7D
Info: dddd plays 9D
S:
C: 7
D: 8 4
H:
[S] play> Info: 123456 won
Info: 123456 plays 8S
Info: CCC plays 9C
Info: dddd plays 8C
S:
C: 7
D: 8
H:
[S] play> Info: 123456 won
Info: 123456 plays JS
Info: CCC plays TD
Info: dddd plays JD
S:
C:
D: 8
H:
[S] play> Info: 123456 won
Info: Team 1=220, Team 2=150
S: A K J 8 5 3 2
C: J 8 5
D:
H: Q 7 3
Info: 123456 bids 6S
Info: CCC bids 7H
Info: dddd passes
[7H] - Bid (or pass)> Info: 123456 passes
Info: CCC plays AC
Info: dddd plays KC
S: A K J 8 5 3 2
C: J 8 5
D:
H: Q 7 3
[C] play> Info: 123456 plays 3C
Info: CCC won
Info: CCC plays TC
Info: dddd plays 4C
S: A K J 8 5 3 2
C: J 8
D:
H: Q 7 3
[C] play> Info: 123456 plays 7C
Info: CCC won
Info: CCC plays KH
Info: dddd plays 5H
S: A K J 8 5 3 2
C: J
D:
H: Q 7 3
[H] play> Info: 123456 plays 2H
Info: CCC won
Info: CCC plays JD
Info: dddd plays 2D
S: A K J 8 5 3 2
C: J
D:
H: Q 7
[D] play> Info: 123456 plays 5D
Info: CCC won
Info: CCC plays 9D
Info: dddd plays 4D
S: A K J 8 5 3
C: J
D:
H: Q 7
[D] play> Info: 123456 plays 7D
Info: CCC won
Info: CCC plays 9H
Info: dddd plays JH
S: A K J 8 5
C: J
D:
H: Q 7
[H] play> Info: 123456 plays AH
Info: 123456 won
Info: 123456 plays QC
Info: CCC plays 2C
Info: dddd plays 9C
S: A K J 8 5
C: J
D:
H: Q
[C] play> Info: 123456 won
Info: 123456 plays 6H
Info: CCC plays 4H
Info: dddd plays 8H
S: A K J 8 5
C:
D:
H: Q
[H] play> Info: zzzzzzzz won
S: A K J 8 5
C:
D:
H:
Lead> Info: 123456 plays QS
Info: CCC plays 6S
Info: dddd plays 4S
Info: 123456 won
Info: 123456 plays 7S
Info: CCC plays TS
Info: dddd plays 9S
S: A K J 8
C:
D:
H:
[S] play> Info: CCC won
Info: CCC plays 6C
Info: dddd plays 8D
S: A K J
C:
D:
H:
[C] play> Info: 123456 plays TD
Info: CCC won
Info: CCC plays 3D
Info: dddd plays AD
S: A K
C:
D:
H:
[D] play> Info: 123456 plays KD
Info: dddd won
Info: dddd plays QD
S: A
C:
D:
H:
[D] play> Info: 123456 plays TH
Info: CCC plays 6D
Info: 123456 won
Info: Team 1=220, Team 2=350
S: 9 4
C: K 9 4
D: A Q 8 4 2
H: J 8 5
Info: 123456 bids 9H
Info: 123456 plays 2S
Info: CCC plays QS
Info: dddd plays TS
S: 9 4
C: K 9 4
D: A Q 8 4 2
H: J 8 5
[S] play> Info: CCC won
Info: CCC plays 3C
Info: dddd plays AC
S: 9
C: K 9 4
D: A Q 8 4 2
H: J 8 5
[C] play> Info: 123456 plays 5C
Info: dddd won
Info: dddd plays KH
S: 9
C: 9 4
D: A Q 8 4 2
H: J 8 5
[H] play> Info: 123456 plays 3H
Info: CCC plays AH
Info: CCC won
Info: CCC plays 7S
Info: dddd plays 6S
S: 9
C: 9 4
D: A Q 8 4 2
H: J 8
[S] play> Info: 123456 plays 3S
Info: zzzzzzzz won
S:
C: 9 4
D: A Q 8 4 2
H: J 8
Lead> Info: 123456 plays 8C
Info: CCC plays 5D
Info: dddd plays JD
Info: zzzzzzzz won
S:
C: 9 4
D: Q 8 4 2
H: J 8
Lead> Info: 123456 plays JC
Info: CCC plays 7C
Info: dddd plays TC
Info: 123456 won
Info: 123456 plays AS
Info: CCC plays KD
Info: dddd plays 6C
S:
C: 4
D: Q 8 4 2
H: J 8
[S] play> Info: zzzzzzzz won
S:
C: 4
D: Q 8 4 2
H: J
Lead> Info: 123456 plays QH
Info: CCC plays 2H
Info: dddd plays 4H
Info: 123456 won
Info: 123456 plays 7H
Info: CCC plays 6H
Info: dddd plays 9H
S:
C: 4
D: Q 8 4 2
H:
[H] play> Info: dddd won
Info: dddd plays 2C
S:
C: 4
D: Q 8 4
H:
[C] play> Info: 123456 plays 5S
Info: CCC plays QC
Info: CCC won
Info: CCC plays 7D
Info: dddd plays 3D
S:
C:
D: Q 8 4
H:
[D] play> Info: 123456 plays 8S
Info: zzzzzzzz won
S:
C:
D: 8 4
H:
Lead> Info: 123456 plays JS
Info: CCC plays TD
Info: dddd plays 9D
Info: CCC won
Info: CCC plays TH
Info: dddd plays 6D
S:
C:
D: 4
H:
[H] play> Info: 123456 plays KS
Info: CCC won
Info: Team 1=-80, Team 2=350
S: 9 4
C: K 7 4
D: A Q 8 4 2
H: J 8 5
Info: 123456 bids 9H
Info: 123456 plays 3H
Info: CCC plays AH
Info: dddd plays 4H
S: 9 4
C: K 7 4
D: A Q 8 4 2
H: J 8 5
[H] play> Info: CCC won
Info: CCC plays 7S
Info: dddd plays TS
S: 9 4
C: K 7 4
D: A Q 8 4 2
H: J 8
[S] play> Info: 123456 plays AS
Info: 123456 won
Info: 123456 plays KS
Info: CCC plays QS
Info: dddd plays 6S
S: 9
C: K 7 4
D: A Q 8 4 2
H: J 8
[S] play> Info: 123456 won
Info: 123456 plays 5C
Info: CCC plays QC
Info: dddd plays 2C
S:
C: K 7 4
D: A Q 8 4 2
H: J 8
[C] play> Info: CCC won
Info: CCC plays KD
Info: dddd plays 3D
S:
C: K 7
D: A Q 8 4 2
H: J 8
[D] play> Info: 123456 plays 7H
Info: 123456 won
Info: 123456 plays QH
Info: CCC plays TH
Info: dddd plays 9H
S:
C: K 7
D: A Q 8 4
H: J 8
[H] play> Info: 123456 won
Info: 123456 plays 2S
Info: CCC plays 2H
Info: dddd plays AC
S:
C: K 7
D: A Q 8 4
H: J
[S] play> Info: CCC won
Info: CCC plays 6H
Info: dddd plays KH
S:
C: K 7
D: A Q 8
H: J
[H] play> S:
C: K 7
D: A Q 8
H: J
[H] play> Info: 123456 plays 3S
Info: dddd won
Info: dddd plays JD
S:
C: K 7
D: A Q 8
H:
[D] play> Info: 123456 plays 5S
Info: CCC plays 5D
Info: zzzzzzzz won
S:
C: K 7
D: Q 8
H:
Lead> Info: 123456 plays JC
Info: CCC plays 3C
Info: dddd plays 8C
Info: 123456 won
Info: 123456 plays 8S
Info: CCC plays 9C
Info: dddd plays 6D
S:
C: K
D: Q 8
H:
[S] play> Info: 123456 won
Info: 123456 plays JS
Info: CCC plays 7D
Info: dddd plays 6C
S:
C:
D: Q 8
H:
[S] play> Info: 123456 won
Info: 123456 plays TC
Info: CCC plays TD
Info: dddd plays 9D
S:
C:
D: Q
H:
[C] play> Info: 123456 won
Info: Team 1=220, Team 2=350
S: A K J 8 5 3 2
C: J 8 5
D:
H: Q 7 3
Info: 123456 bids 9H
Info: 123456 plays AH
Info: CCC plays KH
Info: dddd plays 5H
S: A K J 8 5 3 2
C: J 8 5
D:
H: Q 7 3
[H] play> Info: 123456 won
Info: 123456 plays KD
Info: CCC plays JD
Info: dddd plays 2D
S: A K J 8 5 3 2
C: J 8 5
D:
H: Q 7
[D] play> Info: zzzzzzzz won
S: A K J 8 5 3 2
C: J 8 5
D:
H: Q
Lead> Info: 123456 plays 2H
Info: CCC plays 4H
Info: dddd plays 8H
Info: zzzzzzzz won
S: A K J 8 5 3 2
C: J 8 5
D:
H:
Lead> Info: 123456 plays 7S
Info: CCC plays 6S
Info: dddd plays 4S
Info: zzzzzzzz won
S: K J 8 5 3 2
C: J 8 5
D:
H:
Lead> Info: 123456 plays QS
Info: CCC plays TS
Info: dddd plays 9S
Info: zzzzzzzz won
S: J 8 5 3 2
C: J 8 5
D:
H:
Lead> Info: 123456 plays 6H
Info: CCC plays AC
Info: dddd plays 4C
Info: 123456 won
Info: 123456 plays QC
Info: CCC plays TC
Info: dddd plays 9C
S: 8 5 3 2
C: J 8 5
D:
H:
[C] play> Info: 123456 won
Info: 123456 plays 7C
Info: CCC plays 2C
Info: dddd plays KC
S: 8 5 3 2
C: J 8
D:
H:
[C] play> S: 8 5 3 2
C: J 8
D:
H:
[C] play> Info: dddd won
Info: dddd plays AD
S: 8 5 3 2
C: J
D:
H:
[D] play> Info: 123456 plays TD
Info: CCC plays 9D
Info: dddd won
Info: dddd plays JH
S: 5 3 2
C: J
D:
H:
[H] play> Info: 123456 plays TH
Info: CCC plays 9H
Info: dddd won
Info: dddd plays 4D
S: 5 3
C: J
D:
H:
[D] play> Info: 123456 plays 5D
Info: CCC plays 3D
Info: 123456 won
Info: 123456 plays 7D
Info: CCC plays 6D
Info: dddd plays 8D
S: 5
C: J
D:
H:
[D] play> Info: dddd won
Info: dddd plays QD
S: 5
C:
D:
H:
[D] play> Info: 123456 plays 3C
Info: CCC plays 6C
Info: dddd won
Info: Team 1=520, Team 2=350
Info: Winner is Team 1
